internal/transport/proxy.go

```diff
@@ -72,21 +54,24 @@ func basicAuth(username, password string) string {
 	return base64.StdEncoding.EncodeToString([]byte(auth))
 }
 
-func doHTTPConnectHandshake(ctx context.Context, conn net.Conn, backendAddr string, proxyURL *url.URL, grpcUA string) (_ net.Conn, err error) {
+func doHTTPConnectHandshake(ctx context.Context, conn net.Conn, address resolver.Address, grpcUA string) (_ net.Conn, err error) {
 	defer func() {
 		if err != nil {
 			conn.Close()
 		}
 	}()
 
+	backendAddr := address.Attributes.Value("proxyConnectAddr").(string)
```

Comment:
Calls to `Attributes.Value` with a specific string key should be replaced by calls to a function provided by the `delegatingresolver` package.